### DISCUSSION TITLE
STM32F411CE with ili9486 8 bit parallel mode not working.

### DISCUSSION THREAD

**Discussion by JackDanl** [Bodmer/TFT_eSPI#1104] [General]
#define STM32
#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts
#define SMOOTH_FONT
#define SUPPORT_TRANSACTIONS
#define ILI9486_DRIVER
#define STM_PORTA_DATA_BUS
#define TFT_PARALLEL_8_BIT

platformio.ini

[env:blackpill_f411ce]
platform = ststm32
board = blackpill_f411ce
framework = arduino
lib_deps = 
   TFT_eSPI
   SPI
upload_protocol = dfu
build_flags =
   -D PIO_FRAMEWORK_ARDUINO_ENABLE_CDC
   -D USBCON
   -D USBD_VID=0x0483
   -D USBD_PID=0x5740
   -D USB_MANUFACTURER="Unknown"
   -D USB_PRODUCT="\"BLACKPILL_F411CE\""
   -D HAL_PCD_MODULE_ENABLED
   -D TFT_CS=PB8
   -D TFT_DC=PB7                                  ; Data/Comand pin
   -D TFT_RST=PB9                                ; Reset pin
   -D TFT_D0=PA0
   -D TFT_D1=PA1
   -D TFT_D2=PA2
   -D TFT_D3=PA3
   -D TFT_D4=PA4
   -D TFT_D5=PA5
   -D TFT_D6=PA6
   -D TFT_D7=PA7
   -D TFT_RD=PB0
   -D TFT_WR=PB6

LCD Board wiring 
//  LCD pins  |D7  |D6  |D5 |D4  |D3  |D2  |D1  |D0 | |RD|WR|RS  | CS  |RST|  |SD_SS|SD_DI|SD_DO|SD_SCK|
//STM32 pin |PA7|PA6|PA5|PA4|PA3|PA2|PA1|PA0| |PB0|PB6|PB7|PB8|PB9| |PA15  |PB5     |PB4     |PB3   |

The display works fine using MCUFriend_Kbv. I am not changing any of the wiring, just transferring the settings to TFT_eSPI
MCUFriend detects the display as a ili9486. with 8 bit parallel 
I have pasted my platformio.ini and defines from main.cpp which is just the color_test.ino file

What I am getting is nothing on the display at all, just white. 

Any thoughts on what I am doing wrong?

**Comment by Bodmer** [comment]
See [example here](https://github.com/Bodmer/TFT_eSPI/blob/master/Tools/PlatformIO/Configuring%20options.txt).

Note in particular the line:
  -DUSER_SETUP_LOADED=1

Run the Read_User_Setup diagnostic sketch to see if the compiler is picking up the right settings.

**Reply by JackDanl** [reply]
[code]
TFT_eSPI ver = 2.3.61 
Processor    = Generic
Transactions = Yes    
Interface    = SPI
Display driver = 9486
Display width  = 320
Display height = 480

TFT_CS   = GPIO -1
TFT_DC   = GPIO -1
TFT_RST  = GPIO -1
TFT_WR   = GPIO -1
TFT_RD   = GPIO -1


Display SPI frequency = 20.00
[/code]


Current platformio.ini file
[env:blackpill_f411ce]
platform = ststm32
board = blackpill_f411ce
framework = arduino
lib_deps = 
   TFT_eSPI
   SPI
upload_protocol = dfu
build_flags =
   -D PIO_FRAMEWORK_ARDUINO_ENABLE_CDC
   -D USBCON
   -D USBD_VID=0x0483
   -D USBD_PID=0x5740
   -D USB_MANUFACTURER="Unknown"
   -D USB_PRODUCT="\"BLACKPILL_F411CE\""
   -D HAL_PCD_MODULE_ENABLED
   -D TFT_CS=PB8
   -D TFT_DC=PB7                                  ; Data/Comand pin
   -D TFT_RST=PB9                                ; Reset pin
   -D TFT_D0=PA0
   -D TFT_D1=PA1
   -D TFT_D2=PA2
   -D TFT_D3=PA3
   -D TFT_D4=PA4
   -D TFT_D5=PA5
   -D TFT_D6=PA6
   -D TFT_D7=PA7
   -D TFT_RD=PB0
   -D TFT_WR=PB6
   -D USER_SETUP_LOADED=1
   -D ILI9486_DRIVER=1
   -D TFT_WIDTH=320 
   -D TFT_HEIGHT=480

**Comment by Bodmer** [comment]
Now need to select the parallel interface to the setup list by adding:

-D TFT_PARALLEL_8_BIT=1

**Reply by JackDanl** [reply]
[code]
TFT_eSPI ver = 2.3.61
Processor    = STM32 
Transactions = Yes   
Interface    = Parallel
Display driver = 9486
Display width  = 320
Display height = 480


>>>>> Note: STM32 pin references above D15 may not reflect board markings <<<<<
TFT_CS   = D24
TFT_DC   = D23
TFT_RST  = D25
TFT_WR   = D22
TFT_RD   = D45

TFT_D0   = D37
TFT_D1   = D38
TFT_D2   = D39
TFT_D3   = D40
TFT_D4   = D41
TFT_D5   = D42
TFT_D6   = D43
TFT_D7   = D44


[/code]

platformio.ini 
; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[env:blackpill_f411ce]
platform = ststm32
board = blackpill_f411ce
framework = arduino
lib_deps = 
   TFT_eSPI
   SPI
upload_protocol = dfu
build_flags =
   -D PIO_FRAMEWORK_ARDUINO_ENABLE_CDC
   -D USBCON
   -D USBD_VID=0x0483
   -D USBD_PID=0x5740
   -D USB_MANUFACTURER="Unknown"
   -D USB_PRODUCT="\"BLACKPILL_F411CE\""
   -D HAL_PCD_MODULE_ENABLED
   -D TFT_CS=PB8
   -D TFT_DC=PB7                                  ; Data/Comand pin
   -D TFT_RST=PB9                                ; Reset pin
   -D TFT_D0=PA0
   -D TFT_D1=PA1
   -D TFT_D2=PA2
   -D TFT_D3=PA3
   -D TFT_D4=PA4
   -D TFT_D5=PA5
   -D TFT_D6=PA6
   -D TFT_D7=PA7
   -D TFT_RD=PB0
   -D TFT_WR=PB6
   -D USER_SETUP_LOADED=1
   -D ILI9486_DRIVER=1
   -D TFT_WIDTH=320
   -D TFT_HEIGHT=480
   -D TFT_PARALLEL_8_BIT=1
   -D STM32=1

**Reply by JackDanl** [reply]
Since my last post there has been major progress. 
Seems that as long as I do the defines in the platformio.ini file this work, but when when using #define xxxxxxx within the main.cpp file it does not. Eh, workable.

now I have moved on to displaying jpegs with the TFT_Flash_Jpg example
it is working for the most part. 

https://user-images.githubusercontent.com/14206695/112740076-7004b380-8f3f-11eb-814b-cf5223af8834.mov

When using -D STM_PORTA_DATA_BUS=1 I get the line in the middle of some of the displayed files. Same few lines each time, but it doesn't appear in the others

If I don't include that, there is no line in the middle
